### PR TITLE
Starting with CRUD

### PR DESCRIPTION
**Pull request by bpilegi98**
This is just a commit of the basic CRUD so Carlos can start working on the handler 😄

### CODE REVIEW

File: entity/product.go [hunk 1]
```diff
@@ -1,8 +1,9 @@
-package models
+package entity
 
 type Product struct {
```
Comment (pabloamura) [inline entity/product.go:3]: En realidad en el microservicio de invoices, deberia haber una entidad de invoice, con Id del producto, pero no una tabla (en principio) de productos, sino que cuandos se necesita algun dato del producto, se consume el microservicio de productos. Hagamos ese cambio y despues en todo caso, en otra version, podemos modificarlo por cuestiones de performance, pero en principio el approach microservicio sería asi me parece

**Review by pabloamura — CHANGES_REQUESTED** [review]

File: entity/invoice.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ type Invoice struct {
 	Id          uint      `gorm:"primaryKey"`
 	ClientId    uint      `gorm:"index"`
 	InvoiceDate time.Time ``
-	Products    []Product `json:",omitempty"`
+	Products    []uint    `json:",omitempty"`
```
Comment (pabloamura) [inline entity/invoice.go:11]: Add invoice items or invoice details

**Review by pabloamura — CHANGES_REQUESTED** [review]

File: repository/invoice.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package repository
+
+import (
+	"training-go-invoices/entity"
+
+	"gorm.io/gorm"
+)
+
+//Is this ok?
+var db = &gorm.DB{}
```
Comment (mludeiro) [inline repository/invoice.go:10]: No, you should use the DB already declared in the container. Maybe by injection (creating a repository object). A simpler way is to use a singleton, preferably for the container that has the db associated
Please check out the container definition in go-micro, or we can have a meet to clarify

**Review by mludeiro — CHANGES_REQUESTED** [review]
//Is this ok?
var db = &gorm.DB{}
 
No, you should use the DB defined in the container. Maybe by injection (creating a repository object). A simpler way is to use a singleton, for the db or the container that has the db associated
Please check out the container definition in go-micro

**Review by mludeiro — APPROVED** [review]

**Pull request merged**